dpnp/dpnp_iface_histograms.py

```diff
@@ -306,6 +299,50 @@ def digitize(x, bins, right=False):
     return bins.size - dpnp.searchsorted(bins[::-1], x, side=side)
 
 
+def _result_type_for_device(dtype1, dtype2, device):
+    rt = dpnp.result_type(dtype1, dtype2)
+    return map_dtype_to_device(rt, device)
+
+
+def _supported_by_device(dtype, device):
+    if dtype in [dpnp.float64, dpnp.complex128]:
+        return device.has_aspect_fp64
+
+    if dtype == dpnp.float16:
+        return device.has_aspect_fp16
+
+    return True
+
+
+def _can_cast(dtype1, dtype2, device):
+    if not _supported_by_device(dtype1, device) or not _supported_by_device(
+        dtype2, device
+    ):
+        return False
+
+    return dpnp.can_cast(dtype1, dtype2)
+
+
+def _align_dtypes(a_dtype, bins_dtype, ntype, supported_types, device):
+    a_bin_dtype = _result_type_for_device(a_dtype, bins_dtype, device)
+
+    # histogram implementation doesn't support uint64 as histogram type
+    # we can use int64 instead. Result would be correct even in case of overflow
+    if ntype == numpy.uint64:
+        ntype = dpnp.int64
+
+    if (a_bin_dtype, ntype) in supported_types:
+        return a_bin_dtype, ntype
+
+    for sample_type, hist_type in supported_types:
+        if _can_cast(a_bin_dtype, sample_type, device) and _can_cast(
+            ntype, hist_type, device
+        ):
+            return sample_type, hist_type
+
+    return None, None
```

Comment:
Could you please add a test to cover that line?

Reply:
I can't. This line is unreachable for current configuration of supported types